pkg/sources/git/git.go

```diff
@@ -542,87 +541,107 @@ func (s *Git) ScanCommits(ctx context.Context, repo *git.Repository, path string
 	}
 	logger.V(1).Info("scanning repo", logValues...)
 
-	for commit := range commitChan {
-		if len(scanOptions.BaseHash) > 0 {
-			if commit.Hash == scanOptions.BaseHash {
-				logger.V(1).Info("reached base commit", "commit", commit.Hash[:7])
-				break
-			}
-		}
+	var depth int64
+	var lastCommitHash string
+	for diff := range diffChan {
 		if scanOptions.MaxDepth > 0 && depth >= scanOptions.MaxDepth {
 			logger.V(1).Info("reached max depth", "depth", depth)
 			break
 		}
-		depth++
-		atomic.AddUint64(&s.metrics.commitsScanned, 1)
-		logger.V(5).Info("scanning commit", "commit", commit.Hash[:7])
-		for _, diff := range commit.Diffs {
-			diff := diff
-			if !scanOptions.Filter.Pass(diff.PathB) {
-				continue
-			}
 
-			fileName := diff.PathB
-			if fileName == "" {
-				continue
+		fullHash := diff.CommitHash
+		if !strings.EqualFold(fullHash, lastCommitHash) {
+			depth++
+			lastCommitHash = fullHash
+			atomic.AddUint64(&s.metrics.commitsScanned, 1)
```

Comment:
Is `s.metrics.commitsScanned` shared with other goroutines?

Reply:
Yea i think it is... i'm not entirely sure so I didn't change the exiting behavior jut to be safe.